File: medium/80/80.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        l = 1
        count = 1
        for r in range(1, len(nums)):
            if nums[r] == nums[r - 1]:
                count += 1
            else:
                count = 1
            if count <= 2:
                nums[l] = nums[r]
                l += 1
        return l
# Time Complexity O(n)
# Space Complexity O(1)
            

File: medium/80/80.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int count=1,l=1,r=1;
        while (r<nums.length){
            if (nums[r] == nums[r-1]){
                count++;
            }
            else count=1;
            if (count<=2){
                nums[l] = nums[r];
                l++;
            }
            r++;
        }
        return l;
    }
}

// Time complexity O(n)
// Space complexity O(1)